### ISSUE TITLE
Text align for header does not work

### ISSUE THREAD

**Issue by vogella** [asciidoctor/asciidoctor-pdf#572]
We started to use theming for the pdf output. Really nice feature but heading alignment does not work.
````
heading:
   align: right
````
does not result in heading text aligned to the right.

This does work for the title page:
```` 
title_page:
   align: right
````

**Comment by mojavelinux** [comment]
Are you sure you're using alpha 12 or 13? This issue was resolved in alpha 12. See #343.

If you are still having a problem, can you share your whole theme file?

**Comment by vogella** [comment]
Thanks, for the clarification. I'm still using alpha11, as the Java wrapper has not been released for 12 or 13.

**Comment by mojavelinux** [comment]
To allow you to test alphas quicker, why don't you switch to using the gem directly? AsciidoctorJ PDF just conveniently wraps up Asciidoctor PDF as a jar, but JRuby can still use the native gem.

Here's an example when using Gradle: https://github.com/asciidoctor/asciidoctor-gradle-examples/blob/5f88408e1788562ff363dbb61e2de04f0fd11121/asciidoc-diagram-to-html-example/build.gradle

Just replace asciidoctor-diagram with asciidoctor-pdf (or add it if you're also using asciidoctor-diagram).

```groovy
plugins {
    id 'org.asciidoctor.convert' version '1.5.3'
    id 'com.github.jruby-gradle.base' version '1.3.3'
}

apply plugin: 'org.asciidoctor.convert'

dependencies {
    gems 'rubygems:asciidoctor-pdf:1.5.0.alpha.13'
}

asciidoctorj {
    version = '1.5.4.1'
}

asciidoctor {
    dependsOn jrubyPrepare
    requires = ['asciidoctor-pdf']
    gemPath = jrubyPrepare.outputDir
    backends 'pdf'
}
```

There's no need to wait on AsciidoctorJ for non-core modules.

**Comment by vogella** [comment]
Thanks for the tip with directly using  asciidoctor-pdf. It worked in the example code, but when I tried it in my "normal" document, it failed and I were not able to solve it. Fortunately @robertpanzer helped me to compile locally the j-wrapper, so I'm now using alpha.13.

I think margin-top does not work:
````
heading:
  h1:
   margin_top: 200
   align: right
````
align works now but "margin_top" does not. Should it work?

**Comment by mojavelinux** [comment]
The margin top on heading currently only works when the heading follows other content. It doesn't work when the heading is at the top of the page. Let's open that as a separate issue.

> Thanks for the tip with directly using asciidoctor-pdf. It worked in the example code, but when I tried it in my "normal" document, it failed and I were not able to solve it. 

I'm really curious to know why it failed. That should not be. Can you give me access to the document / repository you're converting.

**Comment by vogella** [comment]
@mojavelinux we have our own little wrapper Gradle plug-in so that we configure the source location, etc. In this plug-in I was not able to change our logic to use asciidoctor-pdf. directly.

Repo: https://github.com/vogellacompany/com.vogella.gradle.plugins/tree/master/com.vogella.gradle.plugin.asciidoc

Document: Anything, use gradlew createPdfBook

But I don't expect you to invest in our build script, I think if necessary we can adjust it, the whole Ruby world is a bit strange for me. IIRC I had the error "svg something required Ruby 2.0.0" and ruby --version showed me ruby "2.3.1p112".

**Comment by mojavelinux** [comment]
Indeed, since you have levels of abstraction, the setup is a bit more nuanced.

> "svg something required Ruby 2.0.0" and ruby --version showed me ruby "2.3.1p112".

It's important to keep in mind that the Ruby on your system is not the Ruby that is being used by AsciidoctorJ. Instead, AsciidoctorJ uses the Ruby version provided by JRuby (via the classpath). By default, the JRuby used by AsciidoctorJ is JRuby 1.7, which is Ruby 1.9.3. The best thing to do, and something I recommend, is to switch to JRuby 9.1.5.0 (which is going to be the default in AsciidoctorJ 1.6.0). But again, since you are using plugins, it might be easier just to wait for the beta release of AsciidoctorJ.

**Comment by vogella** [comment]
> It's important to keep in mind that the Ruby on your system is not the Ruby that is being used by AsciidoctorJ 

Big ah! I was poking around this for a while. At least now this makes sense. Thanks for the clarification.

**Comment by robertpanzer** [comment]
When converting with the asciidoctor-gradle-plugin it might already work to
add a dependency on org.jruby:jruby-complete:9.1.5.0 to the classpath in
the buildscript configuration at the same place where you added the
dependency on asciidoctorj-pdf.

Just a wild guess though.

Am Donnerstag, 22. September 2016 schrieb Dan Allen :

**Comment by mojavelinux** [comment]
Correct. In fact, you can just use a resolutionStrategy stanza to force the version of JRuby that Gradle picks when it performs the jar resolution.

```
buildscript {
  configurations.all {
    resolutionStrategy {
      force 'org.jruby:jruby:9.1.5.0'
    }
  }
}
```

That's the brute force approach. That gives you the latest version of Ruby.

**Comment by vogella** [comment]
Thanks. I keep this bookmarked for the next time I might try this approach. Currently I'm a happy and proud self-build AsciidoctorJ alpha.13 user
